### PR TITLE
README.md for v2.0

### PR DESCRIPTION
**Pull request by gatonero**
Hi Jay!

Here is my first draft of a README.md for v2.0.

Wherever `sensor.smart_irrigation_daily_adjusted_run_time` was previously used, I use `sensor.[zone_name]`. Also for wiki

The screenshots aren't up to date.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,215 +14,225 @@
...
+This component calculates the time to run your irrigation system to compensate for moisture lost by evaporation / evapotranspiration. Using this component you water your garden, lawn or crops precisely enough to compensate what has evaporated. It takes into account precipitation (rain,snow) and adjusts accordingly, so if it rains or snows less or no irrigation is required. Multiple zones can be supported as each zone will have its own flow configuration. 
 
 > **Note - use this component at your own risk - we do not assume responsibility for any inconvience caused by using this component. Always use common sense before deciding to irrigate using the calculations this component provides. For example, irrigating during excessive rainfall might cause flooding. Again - we assume no responsibility for any inconvience caused.**
 
-> **Note If you want to go back and change your settings afterwards, you can either delete the instance and re-create it or edit the entity registry under config/.storage at your own risk.**
+> **Note If you want to go back and change your settings afterwards, you can either delete the zone and re-create it.**
```
Comment (jeroenterheerdt) [inline README.md:22]: we should also add something about the options flow being available where you can enable/disable OWM API, change your API key and API version.

File: README.md [hunk 2]
```diff
@@ -14,215 +14,225 @@
...
+
+The component keeps track of precipitation and at 23:00 (11:00 PM) local time stores it in a daily value.
```
Comment (jeroenterheerdt) [inline README.md:24]: Not entirely. I'd say: "It tracks weather data and uses the weather info at 23:00 (11:00 PM) to calculate duration per zone."

File: README.md [hunk 3]
```diff
@@ -14,215 +14,225 @@
...
+It then calculates the exact runtime in seconds to compensate for the next evaporation.
 
-The component keeps track of hourly precipitation and at 23:00 (11:00 PM) local time stores it in a daily value.
-It then calculates the exact runtime in seconds to compensate for the net evaporation.
 Note that this is the default behavior and this can be disabled if you want more control. Also, the time auto refresh happens (if not disabled) is configurable.
```
Comment (jeroenterheerdt) [inline README.md:27]: there is no auto refresh any more, we have two steps now: auto update and auto calc.

File: README.md [hunk 4]
```diff
@@ -14,215 +14,225 @@
...
 
-This component uses reference evapotranspiration values and calculates base schedule indexes and water budgets from that. This is an industry-standard approach. Information can be found at https://www.rainbird.com/professionals/irrigation-scheduling-use-et-save-water, amongst others.
+This is all the component does, and this is on purpose to provide maximum flexibility. Users are expected to use the value of `sensor.[zone_name]` to interact with their irrigation system and afterwards call the `smart_irrigation.reset_bucket` service. [See the example automations below](#step-4-creating-automation).
+
+This component uses reference evapotranspiration values and calculates water budgets from that. This is an industry-standard approach. Information can be found at https://www.rainbird.com/professionals/irrigation-scheduling-use-et-save-water, amongst others.
+
 The component uses the [PyETo module to calculate the evapotranspiration value (fao56)](https://pyeto.readthedocs.io/en/latest/fao56_penman_monteith.html). Also, please see the [How this works](https://github.com/jeroenterheerdt/HAsmartirrigation/wiki/How-this-component-works) Wiki page.
```
Comment (jeroenterheerdt) [inline README.md:33]: but also other modules are available and can be extended by the community.

File: README.md [hunk 5]
```diff
@@ -14,215 +14,225 @@
...
+
 4. Once a day (time is configurable) the `netto precipitation` is added/substracted from the `bucket,` which starts as empty. If the `bucket` is below zero, irrigation is required. 
-5. Irrigation should be run for `daily_adjusted_run_time`, which is 0 if `bucket`>=0. Afterwards, the `bucket` needs to be reset (using `reset_bucket`). It's up to the user of the component to build the automation for this final step.
+
+5. Irrigation should be run for `sensor.[zone_name]`, which is 0 if `bucket`  >=0. Afterwards, the `bucket` needs to be reset (using `reset_bucket`). It's up to the user of the component to build the automation for this final step.
 
 There are many more options available, see below. To understand how `precipitation`, `netto precipitation`, the `bucket` and irrigation interact, see [example behavior on the Wiki](https://github.com/jeroenterheerdt/HAsmartirrigation/wiki/Example-behavior-in-a-week).
 
 ## Operation modes
```
Comment (jeroenterheerdt) [inline README.md:50]: maybe we can replace this by saying something about mappings? They effectively replace these modes and are more flexible.

File: README.md [hunk 6]
```diff
@@ -14,215 +14,225 @@
...
+
+Since this component provides multiple configuration options it might get confusing about in which scenario what behavior can be expected and what input is required. In the table below we summarize the configuration modes, their accuracy, the required input and how daily run time is calculated. Keep in mind that run time is based on the netto precipitation (precipitation - evapotranspiration) and the bucket value for previous days.1
+|Mode|Accuracy|Input|How adjusted run time is calculated|
 
-Since this component provides multiple configuration options it might get confusing about in which scenario what behavior can be expected and what input is required. In the table below we summarize the configuration modes, their accuracy, the required input and how daily adjusted run time is calculated. Keep in mind that daily adjusted run time is based on the netto precipitation (precipitation - evapotranspiration) and the bucket value for previous days.
 |Mode|Accuracy|Input|How adjusted run time is calculated|
```
Comment (jeroenterheerdt) [inline README.md:67]: "adjusted run time" is not a thing anymore in v2. I'd consider removing this completely and instead talk about mappings / calc modules

File: README.md [hunk 7]
```diff
@@ -14,215 +14,225 @@
...
 |Mode 3 - Mixed|Medium|API key is required for any inputs that have not been provided sensors. All inputs are expected to be point-in-time, *except precipitation*. That sensor is normally provided by a weather station or weather service as a 'total precipitation today' sensor and that is what is expected by the component|If a sensor is provided for precipitation, most recent value for precipitation is used. Otherwise the average of both precipitation and evapotranspiration.|
 |Mode 4 -Not calculating|Very high|In this mode, just two sensors are required: one for precipitation and one for evapotranspiration. Both are expected to be daily accumulative sensors ('total today'), as is normally the case when provided by a weather service or weather station.|Most recent value of both precipitation and evapotranspiration.|
 
+
 ## Configuration
 
 In this section:
```
Comment (jeroenterheerdt) [inline README.md:77]: love this section!

File: README.md [hunk 8]
```diff
@@ -14,215 +14,225 @@
...
+- API Key for Open Weather Map (optional). Only required in mode 1) and 3). See [Getting Open Weater Map API Key](#getting-open-weather-map-api-key) below for instructions.
+
+
+### Step 2: Configuration of Component
 You will need to specify the following:
+
+#### GENERAL
+This page provides global settings.
+- Automatic duration calculation: If enabled set the time of calculation (HH:MM).
+- Automatic weather data update: If enabled specify how often sensor update should happen (minutes, hours, days). Warning: weatherdata update time on or after calculation time!
```
Comment (jeroenterheerdt) [inline README.md:94]: "weatherdata update time should not be on or after calculation time". Also bug #183 will change this - we will do a weather data update at calculation time if there is no weatherdata available yet.

File: README.md [hunk 9]
```diff
@@ -14,215 +14,225 @@
...
+- Automatic weather data update: If enabled specify how often sensor update should happen (minutes, hours, days). Warning: weatherdata update time on or after calculation time! 
+
+#### ZONES
+Specify one or more irrigation zones here. The component calculates irrigation duration per zone, depending on size, throughput, state, module and mapping. A zone can be _disabled_ (so it doesn't do anything), _automatic_ or _manual_. If in automatic, duration is automatically calculated. If in manual, you specify the duration yourself. If disabled, the zone is not included in any calculations.
+
+> **When entering any values in the configuration of this component, keep in mind that the component will expect inches, sq ft, gallons, gallons per minute, or mm, m<sup>2</sup>, liters, liters per minute respectively depending on the settings in Home Assistant (imperial vs metric system).
+
+- Name: The name of your zone, e.g. garden
+- Size: The size of this zone (m<sup>2</sup> or sq ft)
+- Throughput: The flow of this zone (liter/minute or ???/minute)
```
Comment (jeroenterheerdt) [inline README.md:103]: ??? / minute : gallons per minute

File: README.md [hunk 10]
```diff
@@ -14,215 +14,225 @@
...
+- Bucket: You can manipulate this calculated value.
+- Lead Time: In seconds. Time needed to warm up your irrigation system, e.g. time to establish a connection or start a pump etc.
+
+You can Update all zones and Callculate all.
```
Comment (jeroenterheerdt) [inline README.md:107]: Calculate

File: README.md [hunk 11]
```diff
@@ -14,215 +14,225 @@
...
+
+#### MODULES
+Add one or more modules that calculate irrigation duration. Each module comes with its own configuration and can be used to calculate duration for one or more zones. Modules can't be deleted if they are used by one or more zones.
+
+- 0:PyETO: Calculate duration based on the FAO56 calculation from the PyETO library. 
+  - Coastal: If the location you are tracking is situated on or adjacent to coast of a large land mass or anywhere else where air masses are influenced by a nearby water body, enable this setting.
+  - Solrad behaviour: Should solar radiation estimated from temperature or sun hours or disabled 
+  - Forecast days: How many forecast days taken into account
+- 1:Static: 'Dummy' module with a static configurable duration.
+  - Delta: ???
```
Comment (jeroenterheerdt) [inline README.md:129]: delta is the change in the bucket that is returned. so the 'configurable duration' is not entirely correct. Added https://github.com/jeroenterheerdt/HAsmartirrigation/issues/185 to make sure I update it in the code.

File: README.md [hunk 12]
```diff
@@ -14,215 +14,225 @@
...
+
+#### MAPPINGS
+For sensor configuration take care to make sure the unit the component expects is the same as your sensor provides.**
+
 - Names of sensors that supply required measurements (optional). Only required in mode 2) and 3). See [Measurements and Units](https://github.com/jeroenterheerdt/HAsmartirrigation/wiki/Measurements-and-Units) for more information on the measurements and units expected by this component.
```
Comment (jeroenterheerdt) [inline README.md:134]: the linked wiki page needs updates as we now support more units and also have configurable aggregation types.

File: README.md [hunk 13]
```diff
@@ -14,215 +14,225 @@
...
+
 | Service | Description|
 | --- | --- |
-|`smart_irrigation.[instance]_calculate_daily_adjusted_run_time`|Triggers the calculation of daily adjusted run time. Use only if you disabled automatic refresh in the options.|
-|`smart_irrigation.[instance]_calculate_hourly_adjusted_run_time`|Triggers the calculation of hourly adjusted run time. Use for debugging only.|
-|`smart_irrigation.[instance]_enable_force_mode`|Enables force mode.|
-|`smart_irrigation.[instance]_disable_force_mode`|Disables force mode.|
-|`smart_irrigation.]instance]_reset_bucket`|Resets the bucket to 0. Needs to be called after done irrigating (see below).|
-|`smart_irrigation.[instance]_set_bucket`|Sets the bucket to the provided `value`. Use for debugging only.|
+|`Smart Irrigation: calculate_all_zones`|Triggers the calculation of all zones. Use only if you disabled automatic refresh in the options.|
```
Comment (jeroenterheerdt) [inline README.md:152]: all automatic zones.

File: README.md [hunk 14]
```diff
@@ -14,215 +14,225 @@
...
+|`Smart Irrigation: calculate_zone`|Triggers the calculation of one specific zone.|
+|`Smart Irrigation: reset_all_buckets`|Resets all buckets.|
+|`Smart Irrigation: reset_bucket`|Resets one specific bucket.|
+|`Smart Irrigation: set_all_buckets`|Sets all buckets to a specific `value`.|
+|`Smart Irrigation: set_bucket`|Sets a specific bucket to to a specific `value`.|
+|`Smart Irrigation: update_all_zones`|Updates all zones with weather data|
```
Comment (jeroenterheerdt) [inline README.md:158]: all automatic zones.

File: README.md [hunk 15]
```diff
@@ -14,215 +14,225 @@
...
 
 #### Events
 The component uses a number of events internally that you do not need to pay attention to unless you need to debug things. The exception is the `_start` event.
+
 | Event | Description|
 | --- | --- |
-|`[instance]_start`|Fires depending on `daily_adjusted_run_time` value and sunrise. You can listen to this event to optimize the moment you irrigate so your irrigation starts just before sunrise and is completed at sunrise. See below for examples on how to use this.|
+|`[instance]_start`|Fires depending on `sensor.[zone_name]` value and sunrise. You can listen to this event to optimize the moment you irrigate so your irrigation starts just before sunrise and is completed at sunrise. See below for examples on how to use this.|
```
Comment (jeroenterheerdt) [inline README.md:165]: total duration of all zones and sunrise.

File: README.md [hunk 16]
```diff
@@ -14,215 +14,225 @@
...
 |`[instance]_bucketUpd`|Fired when the bucket is calculated. Happens at automatic refresh time or as a result of the `reset_bucket`, `set_bucket` or `calculate_daily_adjusted_run_time` service.|
```
Comment (jeroenterheerdt) [inline README.md:166]: remove

File: README.md [hunk 17]
```diff
@@ -14,215 +14,225 @@
...
 |`[instance]_forceModeTog`|Fired when the force mode is disabled or enabled. Result of calling `enable_force_mode` or `disable_force_mode`|
```
Comment (jeroenterheerdt) [inline README.md:167]: remove

File: README.md [hunk 18]
```diff
@@ -14,215 +14,225 @@
...
+|`Smart Irrigation: calculate_all_zones`|Triggers the calculation of all zones. Use only if you disabled automatic refresh in the options.|
+|`Smart Irrigation: calculate_zone`|Triggers the calculation of one specific zone.|
+|`Smart Irrigation: reset_all_buckets`|Resets all buckets.|
+|`Smart Irrigation: reset_bucket`|Resets one specific bucket.|
+|`Smart Irrigation: set_all_buckets`|Sets all buckets to a specific `value`.|
+|`Smart Irrigation: set_bucket`|Sets a specific bucket to to a specific `value`.|
+|`Smart Irrigation: update_all_zones`|Updates all zones with weather data|
 
 #### Events
 The component uses a number of events internally that you do not need to pay attention to unless you need to debug things. The exception is the `_start` event.
```
Comment (jeroenterheerdt) [inline README.md:161]: remove

File: README.md [hunk 19]
```diff
@@ -14,215 +14,225 @@
...
-|`smart_irrigation.[instance]_set_bucket`|Sets the bucket to the provided `value`. Use for debugging only.|
+|`Smart Irrigation: calculate_all_zones`|Triggers the calculation of all zones. Use only if you disabled automatic refresh in the options.|
+|`Smart Irrigation: calculate_zone`|Triggers the calculation of one specific zone.|
+|`Smart Irrigation: reset_all_buckets`|Resets all buckets.|
+|`Smart Irrigation: reset_bucket`|Resets one specific bucket.|
+|`Smart Irrigation: set_all_buckets`|Sets all buckets to a specific `value`.|
+|`Smart Irrigation: set_bucket`|Sets a specific bucket to to a specific `value`.|
+|`Smart Irrigation: update_all_zones`|Updates all zones with weather data|
 
 #### Events
```
Comment (jeroenterheerdt) [inline README.md:160]: change to "Start event"

File: README.md [hunk 20]
```diff
@@ -14,215 +14,225 @@
...
+
 | Event | Description|
 | --- | --- |
-|`[instance]_start`|Fires depending on `daily_adjusted_run_time` value and sunrise. You can listen to this event to optimize the moment you irrigate so your irrigation starts just before sunrise and is completed at sunrise. See below for examples on how to use this.|
+|`[instance]_start`|Fires depending on `sensor.[zone_name]` value and sunrise. You can listen to this event to optimize the moment you irrigate so your irrigation starts just before sunrise and is completed at sunrise. See below for examples on how to use this.|
 |`[instance]_bucketUpd`|Fired when the bucket is calculated. Happens at automatic refresh time or as a result of the `reset_bucket`, `set_bucket` or `calculate_daily_adjusted_run_time` service.|
 |`[instance]_forceModeTog`|Fired when the force mode is disabled or enabled. Result of calling `enable_force_mode` or `disable_force_mode`|
 |`[instance]_hourlyUpd`|Fired when the hourly adjusted run time is calculated. Happens approximately every hour and when `calculate_hourly_adjusted_run_time` service is called.|
```
Comment (jeroenterheerdt) [inline README.md:168]: remove

File: README.md [hunk 21]
```diff
@@ -14,215 +14,225 @@
...
-|Initial update delay|Delay before first sensor update after reboot. This is useful if using sensors that do not have a status right after reboot.|30|
-|Coastal|If the location you are tracking is situated on or adjacent to coast of a large land mass or anywhere else where air masses are influenced by a nearby water body, enable this setting.|False|
-|Solar Radiation calculation|From v0.0.50 onwards, the component estimates solar radiation using temperature, which seems to be more accurate. If for whatever reason you wanted to revert back to the pre v0.0.50 behavior (which used a estimation of sun hours) disable this.|True|
-
-## Example behavior in a week
+## Example Behavior in a Week
 This [Wiki page](https://github.com/jeroenterheerdt/HAsmartirrigation/wiki/Example-behavior-in-a-week) provides insight into how this component should behave in certain weather conditions. With this you should be able to do a sanity check against your configuration and make sure everything is working correctly.
-## Available services
+
+## Available Services
```
Comment (jeroenterheerdt) [inline README.md:233]: what is the difference between this and the other section about services?

**Review by jeroenterheerdt — CHANGES_REQUESTED** [review]

**Comment by jeroenterheerdt** [comment]
thanks for your input, I left a review!

**Comment by jeroenterheerdt** [comment]
aos, the PR should be made against hte 2.0 branch, not against master...

File: README.md (hunk 1 above)
Comment (gatonero) [reply README.md:23]: Done

File: README.md (hunk 2 above)
Comment (gatonero) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (gatonero) [reply README.md:27]: Done

File: README.md (hunk 5 above)
Comment (gatonero) [reply README.md:52]: Done

File: README.md (hunk 17 above)
Comment (gatonero) [reply README.md:169]: Done

File: README.md (hunk 18 above)
Comment (gatonero) [reply README.md:163]: Done

File: README.md (hunk 20 above)
Comment (gatonero) [reply README.md:170]: Done

File: README.md (hunk 21 above)
Comment (gatonero) [reply README.md:235]: Removed

**Comment by gatonero** [comment]
I#m relative new to git. I will try to PR it against 2.0 branch.

**Pull request closed without merging**